pkg/agent/svid/rotator_test.go

```diff
@@ -121,7 +116,9 @@ func TestRotator(t *testing.T) {
 			case <-time.After(time.Minute):
 				t.Fatal("timed out waiting for the rotator to shut down")
 			case err = <-errCh:
-				require.True(t, errors.Is(err, context.Canceled), "expected %v, not %v", context.Canceled, err)
+				if err != nil {
```

Comment:
I think we should probably change the rotator code to return ctx.Err() when the contexts are done though. The agent layer above this handles that and it makes for a more consistent return code, since RunTasks (used by Run) will return ctx.Err().